From bd99b8f4afecee6d7f8a1ebb22f050bdf732c5d1 Mon Sep 17 00:00:00 2001
From: FrozenCow <frozencow@gmail.com>
Date: Mon, 11 Jan 2016 23:10:47 +0100
Subject: [PATCH 10/12] sepolicy: allow kernel to read storage

This fixes issues where the kernel would need to read and write
files from internal or external storage. More specifically, the
kernel needs these rules for USB mass storage to work correctly.

Change-Id: I8cb0307727bc0c464d5470e55275ad808e748ee0
---
 common/public/su.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/public/su.te b/common/public/su.te
index 9cd6345..473386b 100644
--- a/common/public/su.te
+++ b/common/public/su.te
@@ -64,4 +64,6 @@ userdebug_or_eng(`
   allow system_app superuser_device:sock_file { read write create setattr unlink getattr };
   allow system_app sudaemon:unix_stream_socket { connectto read write setopt ioctl };
   allow system_app superuser_device:dir { create rw_dir_perms setattr unlink };
+
+  allow kernel sudaemon:fd { use };
 ')
-- 
2.7.4

